src/utils.js

```diff
@@ -39,21 +39,23 @@ export function newtown_raphson_method(x0, f, f_prime, tolerance, epsilon, max_i
     for (let iteration = 0; iteration < max_iterations; iteration++) {
         const y = f(x0)
         const y_prime = f_prime(x0)
+        const delta = (y / y_prime) * (1 - iteration / max_iterations)
+        const x1 = x0 - delta
+
+        // console.info(`Iteration ${iteration}: x = ${x0}, f(x) = ${y}, f'(x) = ${y_prime}, delta = ${delta}`)
 
         if (Math.abs(y_prime) < epsilon) {
             break
         }
 
-        const x1 = x0 - (y / y_prime) * (1 - iteration / max_iterations)
-
         if (Math.abs(x1 - x0) <= tolerance) {
             return x1
         }
 
         x0 = x1
     }
 
-    return null
+    return x0
```

Comment:
Hmm... I wonder if we should leave it like that or return null. Or maybe throw exception 🤔